static/js/compose_validate.js

```diff
@@ -191,29 +189,34 @@ export function warn_if_topic_resolved(topic_changed) {
     const stream_name = compose_state.stream_name();
     const message_content = compose_state.message_content();
     const sub = stream_data.get_sub(stream_name);
-    const $resolved_notice_area = $("#compose_resolved_topic");
+    const $compose_banner_area = $("#compose_banners");
 
     if (sub && message_content !== "" && resolved_topic.is_resolved(topic_name)) {
-        if ($resolved_notice_area.html()) {
+        if ($(`#compose_banners .${compose_error.CLASSNAMES.topic_resolved}`).length) {
             // Error is already displayed; no action required.
             return;
         }
 
+        const button_text = settings_data.user_can_move_messages_between_streams()
```

Comment:
This is a preexisting issue, so should get its own commit to fix, but we should be checking `user_can_edit_topic_of_any_message`, not `user_can_move_messages_between_streams`, not resolving a topic does not involve moving messages between streams.

Reply:
done